sdk/state/payment.go

```diff
@@ -188,6 +190,11 @@ func (c *Channel) validatePayment(ca CloseAgreement) (err error) {
 	if !ca.Details.ConfirmingSigner.Equal(c.localSigner.FromAddress()) && !ca.Details.ConfirmingSigner.Equal(c.remoteSigner) {
 		return fmt.Errorf("close agreement confirmer does not match a local or remote signer, got: %s", ca.Details.ConfirmingSigner.Address())
 	}
+
+	// If the close agreement payment amount is incorrect, error.
+	if ca.Details.Balance-c.Balance() != ca.Details.FinalPaymentAmount {
+		return fmt.Errorf("close agreement payment amount is incorrect")
```

Comment:
Whether it is incorrect or not is subjective and sort of alludes to some malice that may not exist. I think we could say the amount is unexpected. We should probably also include the amounts in the error for debugging.
```suggestion
		return fmt.Errorf("close agreement payment amount is unexpected: current balance: %d proposed balance: %d payment amount: %d")
```